File: backend/src/Broker.Backoffice.Domain/Identity/Permissions.cs
namespace Broker.Backoffice.Domain.Identity;

public static class Permissions
{
    public const string UsersRead = "users.read";
    public const string UsersCreate = "users.create";
    public const string UsersUpdate = "users.update";
    public const string UsersDelete = "users.delete";

    public const string RolesRead = "roles.read";
    public const string RolesCreate = "roles.create";
    public const string RolesUpdate = "roles.update";
    public const string RolesDelete = "roles.delete";

    public const string PermissionsRead = "permissions.read";

    public const string AuditRead = "audit.read";

    public const string ClientsRead = "clients.read";
    public const string ClientsCreate = "clients.create";
    public const string ClientsUpdate = "clients.update";
    public const string ClientsDelete = "clients.delete";

    public const string AccountsRead = "accounts.read";
    public const string AccountsCreate = "accounts.create";
    public const string AccountsUpdate = "accounts.update";
    public const string AccountsDelete = "accounts.delete";

    public const string InstrumentsRead = "instruments.read";
    public const string InstrumentsCreate = "instruments.create";
    public const string InstrumentsUpdate = "instruments.update";
    public const string InstrumentsDelete = "instruments.delete";

    public const string SettingsManage = "settings.manage";

    public static readonly (string Code, string Name, string Group)[] All =
    [
        (UsersRead, "View users", "Users"),
        (UsersCreate, "Create users", "Users"),
        (UsersUpdate, "Update users", "Users"),
        (UsersDelete, "Delete users", "Users"),
        (RolesRead, "View roles", "Roles"),
        (RolesCreate, "Create roles", "Roles"),
        (RolesUpdate, "Update roles", "Roles"),
        (RolesDelete, "Delete roles", "Roles"),
        (PermissionsRead, "View permissions", "Permissions"),
        (AuditRead, "View audit log", "Audit"),
        (ClientsRead, "View clients", "Clients"),
        (ClientsCreate, "Create clients", "Clients"),
        (ClientsUpdate, "Update clients", "Clients"),
        (ClientsDelete, "Delete clients", "Clients"),
        (AccountsRead, "View accounts", "Accounts"),
        (AccountsCreate, "Create accounts", "Accounts"),
        (AccountsUpdate, "Update accounts", "Accounts"),
        (AccountsDelete, "Delete accounts", "Accounts"),
        (InstrumentsRead, "View instruments", "Instruments"),
        (InstrumentsCreate, "Create instruments", "Instruments"),
        (InstrumentsUpdate, "Update instruments", "Instruments"),
        (InstrumentsDelete, "Delete instruments", "Instruments"),
        (SettingsManage, "Manage reference data settings", "Settings"),
    ];
}

File: frontend/apps/backoffice/src/test/msw/handlers/permissions.ts
import { http, HttpResponse } from "msw";
import type { PermissionDto } from "@/api/types";

const BASE = "/api/v1";

const permissions: PermissionDto[] = [
  { id: "p1", code: "users.read", name: "Read Users", description: null, group: "Users" },
  { id: "p2", code: "users.create", name: "Create Users", description: null, group: "Users" },
  { id: "p3", code: "users.update", name: "Update Users", description: null, group: "Users" },
  { id: "p4", code: "users.delete", name: "Delete Users", description: null, group: "Users" },
  { id: "p5", code: "roles.read", name: "Read Roles", description: null, group: "Roles" },
  { id: "p6", code: "roles.create", name: "Create Roles", description: null, group: "Roles" },
  { id: "p7", code: "roles.update", name: "Update Roles", description: null, group: "Roles" },
  { id: "p8", code: "roles.delete", name: "Delete Roles", description: null, group: "Roles" },
  { id: "p9", code: "clients.read", name: "Read Clients", description: null, group: "Clients" },
  { id: "p10", code: "clients.create", name: "Create Clients", description: null, group: "Clients" },
  { id: "p11", code: "clients.update", name: "Update Clients", description: null, group: "Clients" },
  { id: "p12", code: "clients.delete", name: "Delete Clients", description: null, group: "Clients" },
  { id: "p13", code: "audit.read", name: "Read Audit", description: null, group: "Audit" },
  { id: "p14", code: "permissions.read", name: "Read Permissions", description: null, group: "Permissions" },
];

export const permissionsHandlers = [
  http.get(`${BASE}/permissions`, () =>
    HttpResponse.json(permissions),
  ),
];
